tutorials/Building a Custom Environment in Aviary.ipynb

```diff
@@ -1,5 +1,70 @@
 {
  "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "ddcb8adc-f12d-42e2-909b-93ffed6e4259",
+   "metadata": {},
+   "source": [
+    "# Tutorial: Building a Custom Environment in Aviary\n",
+    "\n",
+    "## Pre-requisites\n",
+    "\n",
+    "To get started, you’ll need to install `Aviary`, `LDP`, and `pydantic`, all available on PyPI. Note that `Aviary` is listed on PyPI as `fhaviary`. Run the command below to install these packages:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "73d00726-8059-4ddc-9aef-d2de7d201751",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install fhaviary ldp pydantic"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bd798465-7824-4543-8816-9799542006ac",
+   "metadata": {},
+   "source": [
+    "You will need to set the `OPENAI_API_KEY` or the corresponding key for any other API you wish to access."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bc7b46a9-7fc8-4a75-b455-64b1ffe8d902",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "os.environ[\"OPENAI_API_KEY\"] = \"your_API_key\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9c5fb93f-4ca1-4f7d-bbd3-c3589d0eeb5c",
+   "metadata": {},
+   "source": [
+    "## Background\n",
+    "\n",
+    "[Aviary](https://github.com/Future-House/aviary) is our framework supporting diverse language environments, where actions are tools available to agents. [LDP](https://github.com/Future-House/ldp) is our framework for creating and training language agents.\n",
+    "\n",
+    "Below, we briefly define some key classes and concepts from these libraries for context:\n",
+    "\n",
+    "**From Aviary**  \n",
+    "- `Message`: Used by language agents and environments for communication. Messages include attributes like `content` ot `role` (`system`, `user`, `assistant`, `tool` ), matching OpenAI's conventions.\n",
+    "- `Environment`: An environment is a stateful system or \"world\" where an agent operates by taking actions. In Aviary, these actions are called tools. The environment presents states that the agent observes (totally or partially), prompting it to use tools to affect outcomes. Each action taken yields a reward and leads to a new state.\n",
+    "- `Tool`: Defines an environmental tool that an agent can use to accomplish its task. Different environments offer different tools. Most tools take arguments and multiple tools can be called in parallel.\n",
+    "- `ToolRequestMessage`: This is a specialized subclasses of `Message` used for tool requests. Typically, a language agent sends a `ToolRequestMessage` to the environment to request the execution of a specific tool. The role of `ToolRequestMessage` is always `assistant`.\n",
+    "\n",
+    "**From LDP**   \n",
```

Comment:
```suggestion
    "**From LDP**\n",
```

Can you remove trailing spaces

Reply:
Thank you very much for the quick feedback, @jamesbraza! 😃